library/std/src/keyword_docs.rs

```diff
@@ -14,6 +14,16 @@
 /// assert_eq!(true as u8 + thing2 as u8, 100);
 /// ```
 ///
+/// `as` can also be used with the `_` placeholder when the destination type
+/// can be inferred. Note that this can cause inference breakage and usually
+/// such code should use an explicit type for both clarity and stability:
+///
+/// ```rust
+/// let value1: i32 = 42;
+/// let value2: i64 = value1 as _;
+/// assert_eq!(value2, value1 as _);
```

Comment:
If we're going to document `as _`, I'd rather document `as *const _`/`as *mut _` which show up a lot more often. Maybe also link to https://doc.rust-lang.org/std/primitive.pointer.html#method.cast and mention that it's recommended over `*const _`.